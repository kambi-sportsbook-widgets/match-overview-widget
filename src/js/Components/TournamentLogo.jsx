import React from 'react';
import PropTypes from 'prop-types';
import styles from './TournamentLogo.scss';

const TournamentLogo = ({ logoName }) => {
   return (
      <div className={styles.general}>
         <i
            className='kw-custom-logo-large-type'
            style={{
               backgroundImage: `url(assets/icons/${logoName}.svg)`
            }}
         />
      </div>
   );
}

TournamentLogo.propTypes = {
   /**
    * Logo CSS class name.
    * These classes are defined in operator-specific CSS file.
    */
   logoName: PropTypes.string
};

TournamentLogo.defaultProps = {
   logoName: 'football'
};

export default TournamentLogo;
