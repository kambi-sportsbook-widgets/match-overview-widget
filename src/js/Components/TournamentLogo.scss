@import '~kambi-widget-core-library/src/scss/core.scss';

:local(.general) {
   display: flex;
   align-items: center;
   height: 120px;

   i {
      background: no-repeat 50%;
      background-size: contain;
      height: 90px;
      width: 90px;

      @media (max-width: $screen-sm) {
         height: 80px;
         width: 80px;
      }
   }
}
