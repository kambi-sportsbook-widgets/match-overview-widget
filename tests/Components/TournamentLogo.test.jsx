/* eslint-env jest */
import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import TournamentLogo from '../../src/js/Components/TournamentLogo';

let renderer;

describe('Event DOM rendering', () => {

   beforeEach(() => {
      renderer = new ReactShallowRenderer();
   });

   it('renders correctly with default props', () => {
      expect(renderer.render(
         <TournamentLogo />
      )).toMatchSnapshot();
   });

   it('renders correctly with custom logo CSS class name', () => {
      expect(renderer.render(
         <TournamentLogo logoClassName="test_logo_class_name" />
      )).toMatchSnapshot();
   });

});
