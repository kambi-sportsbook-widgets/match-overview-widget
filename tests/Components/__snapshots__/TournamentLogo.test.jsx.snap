// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event DOM rendering renders correctly with custom logo CSS class name 1`] = `
<div
  className="general"
>
  <i
    className="kw-custom-logo-large-type"
    style={
      Object {
        "backgroundImage": "url(assets/icons/football.svg)",
      }
    }
  />
</div>
`;

exports[`Event DOM rendering renders correctly with default props 1`] = `
<div
  className="general"
>
  <i
    className="kw-custom-logo-large-type"
    style={
      Object {
        "backgroundImage": "url(assets/icons/football.svg)",
      }
    }
  />
</div>
`;
